iis-common/src/main/java/eu/dnetlib/iis/common/report/ReportParamJsonAppender.java

```diff
@@ -0,0 +1,95 @@
+package eu.dnetlib.iis.common.report;
+
+import java.util.List;
+
+import org.apache.hadoop.util.StringUtils;
+
+import com.google.common.collect.Lists;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+import eu.dnetlib.iis.common.schemas.ReportParam;
+
+/**
+ * Appender of {@link ReportParam} into json report.
+ * 
+ * @author madryk
+ */
+public class ReportParamJsonAppender {
+
+
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Appends {@link ReportParam} into the passed json report.<br/>
+     * 
+     * The key of the report param ({@link ReportParam#getKey()}) defines
+     * where the value ({@link ReportParam#getValue()}) should be inserted
+     * in report json structure.<br/>
+     * The key is splitted by dots. Resulting values defines subsequent
+     * fields in json.<br/>
+     * For example:<br/>
+     * <code>new ReportParam("param1.paramA", "34")</code><br/>
+     * will result in following json:<br/>
+     * <code>{"param1": {"paramA": 34}}</code><br/>
+     * If report param key collides with passed json, then any
+     * conflicting node will be replaced with new report param
+     * value.
+     */
+    public void appendReportParam(JsonObject jsonReport, ReportParam reportParam) {
+        
+        String[] jsonFieldHierarchy = StringUtils.split(reportParam.getKey().toString(), '.');
+        
+        insertValue(jsonReport, Lists.newArrayList(jsonFieldHierarchy), convertParamValueToJson(reportParam));
+        
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    private JsonElement convertParamValueToJson(ReportParam reportParam) {
+        
+        long value = Long.valueOf(reportParam.getValue().toString());
```

Comment:
I agree that this may be a little misleading.
But in future, it can be helpful.
If someone would like to use further the generated report. He would know that in this specific field he will always get a numeric value and would not have to parse this value.
I used `string` in `ReportParam.value` to be able to handle not only counters. When we will have different types of report params. For example times of workflows.
We could add a type enum field to ReportParam. And decode ReportParam.value based on this type.